Fix candidates and PA

thierryvolpiatto opened the pull request:
This fix PA and simplify code.

alphapapa commented:
Hi Thierry,

1.  The persistent action worked fine for me before this PR.  Maybe my Helm version is too old?

2.  I think this PR makes the summaries less useful, as most of them now are only one or two sentences.  The code before this PR uses the Wikipedia "extracts" API, which I set up in the Wikipedia API sandbox, and AFAICT it's intended for the purpose of summarizing pages.  It usually provides a paragraph or two, which easily fits on the screen, and provides enough information that, most of the time, I don't need to visit the full article.  This PR seems to make the summaries more like dictionary definitions, which are too brief, IMO.

What do you think?

Thanks.

thierryvolpiatto commented:
alphapapa <notifications@github.com> writes:

> Hi Thierry,
>
>  1. The persistent action worked fine for me before this PR. Maybe my Helm version is too old?

No, the problem was with curl when using `helm-net-prefer-curl` it is
now fixed in helm-net with helm-net-curl-switches.

>  2. I think this PR makes the summaries less useful, as most of them now are only one or two sentences. The code before this PR uses the Wikipedia
>     "extracts" API, which I set up in the Wikipedia API sandbox, and AFAICT it's intended for the purpose of summarizing pages. It usually provides a
>     paragraph or two, which easily fits on the screen, and provides enough information that, most of the time, I don't need to visit the full article. This
>     PR seems to make the summaries more like dictionary definitions, which are too brief, IMO.
>    
> What do you think?

Definitely yes, the output is more complete.

BTW I have fixed the re-search-forward call to avoid error and improved
header as well, feel free to add your copyright if you want.

Now working fine for me with and without curl usage.

Thanks.

-- 
Thierry

alphapapa commented:
> BTW I have fixed the re-search-forward call to avoid error

I actually left it that way on purpose, so it would raise an error if the HTTP request can't be parsed correctly rather than failing silently.  I'd rather users report a bug than think there are no results available or that the tool simply doesn't work sometimes.  Ideally we wouldn't have to skip the HTTP headers manually, but Emacs network libraries are so primitive.  :(  Anyway, what do you think?

> and improved header as well, feel free to add your copyright if you want. 

I've only contributed a few lines, so I think it's fine as is.  :)

> Now working fine for me with and without curl usage. 

Good.  So does this PR still require merging?

Thanks.

thierryvolpiatto commented:
alphapapa <notifications@github.com> writes:

>     BTW I have fixed the re-search-forward call to avoid error
>
> I actually left it that way on purpose, so it would raise an error if
> the HTTP request can't be parsed correctly rather than failing
> silently. I'd rather users report a bug than think there are no
> results available or that the tool simply doesn't work
> sometimes. Ideally we wouldn't have to skip the HTTP headers manually,
> but Emacs network libraries are so primitive. :( Anyway, what do you
> think?

We have to attempt to skip HTML headers, but not throw an error because
the CURL output with -L option have no HTML header i.e. in this case the
search-forward is not needed.

> Good. So does this PR still require merging?

No.

Thanks.

--
Thierry
